### DISCUSSION TITLE
Theming: Loop music with intro / delayed starting point

### DISCUSSION THREAD

**Discussion by ValVoice** [stepmania/stepmania#2134] [Q&A]
I want to loop menu music in my theme, but it has an intro before the first loop. Is there a way to specify the loop offset in milliseconds in the file name, instead of cutting out the intro?

**Comment by SpoOkyMagician** [comment]
A simple solution would just to be to edit the music file itself at
the end and give it some silence. (As much as needed. That should be
enough. You could use audacity or FL Studios to do this with ease.)

**Reply by ValVoice** [reply]
How would adding silence at the end cause the music to loop seamlessly?

If you know music notation, I'm trying to do a *dal segno* instead of a *da capo*.

[The BRSTM file format](https://smashboards.com/threads/how-to-make-looping-a-much-easier-task.306506/) lets me specify a loop start and end point in samples.

**Reply by SpoOkyMagician** [reply]
Oh sorry, I misread your OP... I thought you were just wanting a simple delay while it loops.

edit: If I am understanding you correctly, you want the intro to play once and the music itself loop afterwords?

edit: In that case... Umm... You could have two separate sound actors (which the intro part is played. It would have to be cut in its own file.) Then once that's done, you would use the looping music you want. (Again, that would require cutting parts of it out... If you don't want that... then... I'll have to check the Lua stuff... But, I am not familiar with anything like that off hand.)

edit: Yeah, I don't see anything specific sorry. I believe I know what you are referring to. I honestly never really used it myself but, I am aware of it in FLS. I don't know if SM even supports it though. (and if you are wondering, music is just a hobby of mine. I am no expert.)
